specification/indices/_types/IndexSettings.ts

```diff
@@ -84,11 +84,13 @@ export class IndexSettings
    * @server_default 1
    * @availability stack
    * */
+  // eslint-disable-next-line es-spec-validator/no-inline-unions -- TODO: create named alias
   number_of_shards?: integer | string // TODO: should be only int
```

Comment:
For the record, this a known issue with index settings that they can be passed as strings too.